src/composable/ConditionalOrder.ts

```diff
@@ -287,7 +288,12 @@ export abstract class ConditionalOrder<D, S> {
 
       let orderBookApi = orderBookCache[chainId]
       if (!orderBookApi) {
-        orderBookApi = new OrderBookApi({ ...orderbookApiConfig, chainId })
+        if (orderBookOrConfig && 'context' in orderBookOrConfig) {
+          orderBookApi = orderBookOrConfig as OrderBookApi
```

Comment:
this is a bit fragile. In general we should not use a `as`

Maybe this helps? https://www.typescriptlang.org/docs/handbook/2/narrowing.html#instanceof-narrowing

Reply:
N/A, only using instantiated `OrderBookApi` now